src/uucore/src/lib/parser/parse_size.rs

```diff
@@ -0,0 +1,324 @@
+        Err(_) => return Err(ParseSizeError::size_too_big(size)),
+    };
+    number
+        .checked_mul(factor)
+        .ok_or_else(|| ParseSizeError::size_too_big(size))
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ParseSizeError {
+    ParseFailure(String), // Syntax
+    SizeTooBig(String),   // Overflow
+}
+
+impl Error for ParseSizeError {
+    fn description(&self) -> &str {
+        match *self {
+            ParseSizeError::ParseFailure(ref s) => &*s,
+            ParseSizeError::SizeTooBig(ref s) => &*s,
+        }
+    }
+}
+
+impl fmt::Display for ParseSizeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        let s = match self {
+            ParseSizeError::ParseFailure(s) => s,
+            ParseSizeError::SizeTooBig(s) => s,
+        };
+        write!(f, "{}", s)
+    }
+}
+
+impl ParseSizeError {
+    fn parse_failure(s: &str) -> ParseSizeError {
+        // stderr on linux (GNU coreutils 8.32)
+        // has to be handled in the respective uutils because strings differ, e.g.:
+        //
+        // `NUM`
+        // head:     invalid number of bytes: ‘1fb’
+        // tail:     invalid number of bytes: ‘1fb’
+        //
+        // `SIZE`
+        // split:    invalid number of bytes: ‘1fb’
+        // truncate: Invalid number: ‘1fb’
+        //
+        // `MODE`
+        // stdbuf:   invalid mode ‘1fb’
+        //
+        // `SIZE`
+        // sort:     invalid suffix in --buffer-size argument '1fb'
+        // sort:     invalid --buffer-size argument 'fb'
+        //
+        // `SIZE`
+        // du:       invalid suffix in --buffer-size argument '1fb'
+        // du:       invalid suffix in --threshold argument '1fb'
+        // du:       invalid --buffer-size argument 'fb'
+        // du:       invalid --threshold argument 'fb'
+        //
+        // `BYTES`
+        // od:       invalid suffix in --read-bytes argument '1fb'
+        // od:       invalid --read-bytes argument  argument 'fb'
+        //                   --skip-bytes
+        //                   --width
+        //                   --strings
+        // etc.
+        ParseSizeError::ParseFailure(format!("‘{}’", s))
+    }
+
+    fn size_too_big(s: &str) -> ParseSizeError {
+        // stderr on linux (GNU coreutils 8.32)
+        // has to be handled in the respective uutils because strings differ, e.g.:
+        //
+        // head:     invalid number of bytes: ‘1Y’: Value too large for defined data type
+        // tail:     invalid number of bytes: ‘1Y’: Value too large for defined data type
+        // split:    invalid number of bytes: ‘1Y’: Value too large for defined data type
+        // truncate:          Invalid number: ‘1Y’: Value too large for defined data type
+        // stdbuf:               invalid mode ‘1Y’: Value too large for defined data type
+        // sort:     -S argument '1Y' too large
+        // du:       -B argument '1Y' too large
+        // od:       -N argument '1Y' too large
```

Comment:
Wow, impressive work tracking this all down! This is definitely something that I have to keep in mind for the error handling PR.

Reply:
Yeah, I was surprised at first that they don't all behave the same. But I guess it makes sense if you take the historic aspect into account. My conclusion is that nothing can be taken for granted and every assumption needs to be double checked.